Assets/Plugins/Source/Editor/Build/PlatformSpecificBuilder.cs

```diff
@@ -241,19 +242,37 @@ private static void ConfigureVersion()
         /// </summary>
         private static async void AutoSetProductVersion()
         {
-            var eosConfig = await Config.GetAsync<EOSConfig>();
-            var prebuildConfig = await Config.GetAsync<PrebuildConfig>();
-            var previousProdVer = eosConfig.productVersion;
+            PrebuildConfig prebuildConfig = await Config.GetAsync<PrebuildConfig>();
 
-            if (prebuildConfig.useAppVersionAsProductVersion)
+            // If the product version is set by config, then stop here.
+            if (!prebuildConfig.useAppVersionAsProductVersion)
             {
-                eosConfig.productVersion = Application.version;
+                return;
             }
 
-            if (previousProdVer != eosConfig.productVersion)
+            if (!Version.TryParse(Application.version, out Version newVersion))
             {
-                await eosConfig.WriteAsync(true);
+                Debug.LogWarning(
```

Comment:
We should be able to support a string value for `ProductVersion`, not just a type `Version`.

Reply:
This has been addressed in another PR that is currently open. Once that PR is approved, I will mark this conversation as resolved and move this PR from draft to ready for review.

Reply:
That's been done. Marking conversation as resolved.